string file_path
---
Graph map_graph